libbeat/otelbeat/oteltranslate/tls_otel.go

```diff
@@ -0,0 +1,156 @@
+			"insecure": true,
+		}, nil
+	}
+
+	// throw error if unsupported tls config is passed
+	if err := validateUnsupportedConfig(tlscfg); err != nil {
+		return nil, err
+	}
+
+	if tlscfg.VerificationMode == tlscommon.VerifyNone {
+		insecureSkipVerify = true
+	}
+
+	// unpacks -> ssl.certificate_authorities
+	// The OTel exporter accepts either single CA file or CA string. However,
+	// Beats support any combination and number of files and certificates
+	// as string, so we read them all and assemble one PEM string with
+	// all CA certificates
+	var caCerts []string
+	for _, ca := range tlscfg.CAs {
+		d, err := tlscommon.ReadPEMFile(logger, ca, "")
+		if err != nil {
+			logger.Errorf("Failed reading CA: %+v", err)
+			return nil, err
+		}
+		caCerts = append(caCerts, string(d))
+	}
+	// We only include the system certificates if no CA is defined
+	includeSystemCACertsPool := len(caCerts) == 0
+
+	// TODO: Check if certificate.key is set before attempting to read from it
+	// unpacks ->  ssl.key
+	certKeyBytes, err := tlscommon.ReadPEMFile(logger, tlscfg.Certificate.Key, tlscfg.Certificate.Passphrase)
+	if err != nil {
+		return nil, fmt.Errorf("failed reading key file: %w", err)
+	}
+	certKeyPem := string(certKeyBytes)
+
+	// TODO: Check is certificate.certificate is set before attempting to read from it
+	// unpacks ->  ssl.certificate
+	certBytes, err := tlscommon.ReadPEMFile(logger, tlscfg.Certificate.Certificate, "")
+	if err != nil {
+		logger.Errorf("Failed reading cert file: %+v", err)
+		return nil, fmt.Errorf("failed reading cert file: %w", err)
+	}
+	certPem := string(certBytes)
+
+	tlsConfig, err := tlscommon.LoadTLSConfig(tlscfg)
+	if err != nil {
+		return nil, fmt.Errorf("cannot load SSL configuration: %w", err)
+	}
+	goTLSConfig := tlsConfig.ToConfig()
+
+	// unpacks -> ssl.cipher_suites
+	ciphersuites := []string{}
+	for _, cs := range goTLSConfig.CipherSuites {
+		ciphersuites = append(ciphersuites, tls.CipherSuiteName(cs))
+	}
+
+	otelTLSConfig := map[string]any{
+		"insecure_skip_verify": insecureSkipVerify, // ssl.verirication_mode,
+
+		// Config
+		"include_system_ca_certs_pool": includeSystemCACertsPool,
+		"ca_pem":                       strings.Join(caCerts, ""), // ssl.certificate_authorities
+		"cert_pem":                     certPem,                   // ssl.certificate
+		"key_pem":                      certKeyPem,                // ssl.key
+		"cipher_suites":                ciphersuites,              // ssl.cipher_suites
+	}
+
+	// For type safety check only
+	// the returned valued should match `clienttls.Config` type.
+	// it throws an error if non existing key name is used in the returned map structure
+	var result configtls.ClientConfig
+	d, _ := mapstructure.NewDecoder(&mapstructure.DecoderConfig{
+		Squash:      true,
+		Result:      &result,
+		ErrorUnused: true,
+	})
+
+	if err := d.Decode(otelTLSConfig); err != nil {
+		return nil, err
+	}
+
+	// Also validate the translated config
```

Comment:
The `Validate` method here does not validate if the `cert_pem` and `key_pem` are set given `tls.insecure` is false. 


Created an issue upstream for the same. https://github.com/open-telemetry/opentelemetry-collector/issues/11825